### PR TITLE
Bcftools filter

### PR DESCRIPTION
**Pull request by EmmaCAndersson**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/variant_calling/call_snv_sentieon.nf [hunk 1]
```diff
@@ -38,6 +39,9 @@ workflow CALL_SNV_SENTIEON {
             ch_versions = ch_versions.mix(SENTIEON_DNAMODELAPPLY.out.versions.first())
         }
 
+        BCFTOOLS_FILTER ( ch_vcf )
+        ch_vcf   = BCFTOOLS_FILTER.out.vcf
+
```
[the remark this answers is not in the document]
Comment (sima-r) [reply subworkflows/local/variant_calling/call_snv_sentieon.nf:44]: The version of bcftools_filter is to be added in the ch_version channel

File: conf/modules.config [hunk 2]
```diff
@@ -412,6 +412,14 @@ process {
         ]
     }
 
+    withName: '.*CALL_SNV:CALL_SNV_SENTIEON:BCFTOOLS_FILTER' {
```
Comment (sima-r) [inline conf/modules.config:415]: Change the name here according to what you name the alias in the subworkflow

Comment (sima-r) [reply conf/modules.config:415]: Add : ext.args = "-s 'ML_FAIL' -i 'INFO/ML_PROB <= 0.95' -m x"

File: conf/modules.config [hunk 3]
```diff
@@ -412,6 +412,14 @@ process {
...
+        publishDir = [
+            path: { "${params.outdir}/bcftools_filter" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+
```
Comment (sima-r) [inline conf/modules.config:422]: Add the same section for the second alias of bcftools_filter that you make in the subworkflow with the following arguments:
ext.args = '-i "'FILTER=\"PASS\"'" -m x'  #We need to make sure later if strings get parsed correctly.

File: conf/modules.config [hunk 4]
```diff
@@ -412,6 +412,25 @@ process {
...
+        publishDir = [
+            path: { "${params.outdir}/bcftools_filter" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+        ]
+    }
+    withName: '.*CALL_SNV:CALL_SNV_SENTIEON:BCFTOOLS_FILTER_TWO' {
+         '-i "'FILTER="PASS"'" -m x' #We need to make sure later if strings get parsed correctly
```
Comment (sima-r) [inline conf/modules.config:426]: ext.args =  '-i "'FILTER="PASS"'" -m x'

File: conf/modules.config [hunk 5]
```diff
@@ -412,6 +412,25 @@ process {
...
 
+    withName: '.*CALL_SNV:CALL_SNV_SENTIEON:BCFTOOLS_FILTER_ONE' {
+        ext.args = "-s 'ML_FAIL' -i 'INFO/ML_PROB <= 0.95' -m x"  
+        publishDir = [
+            path: { "${params.outdir}/bcftools_filter" },
+            mode: params.publish_dir_mode,
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+        ]
```
Comment (Gwennid) [inline conf/modules.config:423]: Remove extra set of brackets (rows 423 and 424)

File: conf/modules.config [hunk 6]
```diff
@@ -412,6 +412,25 @@ process {
         ]
     }
 
+    withName: '.*CALL_SNV:CALL_SNV_SENTIEON:BCFTOOLS_FILTER_ONE' {
+        ext.args = "-s 'ML_FAIL' -i 'INFO/ML_PROB <= 0.95' -m x"
```
Comment (Gwennid) [inline conf/modules.config:416]: Remove trailing whitespaces in the end of the row

File: subworkflows/local/variant_calling/call_snv_sentieon.nf [hunk 7]
```diff
@@ -38,6 +40,9 @@ workflow CALL_SNV_SENTIEON {
             ch_versions = ch_versions.mix(SENTIEON_DNAMODELAPPLY.out.versions.first())
         }
 
+        BCFTOOLS_FILTER_ONE ( ch_vcf )
```
Comment (Gwennid) [inline subworkflows/local/variant_calling/call_snv_sentieon.nf:43]: You imported the module as `BCF_FILTER_ONE`...

File: subworkflows/local/variant_calling/call_snv_sentieon.nf [hunk 8]
```diff
@@ -47,6 +52,9 @@ workflow CALL_SNV_SENTIEON {
             .groupTuple()
             .set{ ch_vcf_idx_case }
 
+        BCFTOOLS_FILTER_TWO ( ch_vcf )
```
Comment (Gwennid) [inline subworkflows/local/variant_calling/call_snv_sentieon.nf:55]: Same here - the module is imported as `BCF_FILTER_TWO`

**Comment by jemten** [comment]
@EmmaCAndersson, should we close this PR? Looks like you've added the filtering in #280 also

**Pull request closed without merging**